client/client.go

```diff
@@ -81,6 +82,17 @@ type Client struct {
 	customHTTPHeaders map[string]string
 	// manualOverride is set to true when the version was set by users.
 	manualOverride bool
+
+	// autoVersionNegotiation indicates if the client should automatically negotiate
+	// the API version to use when making requests. API version negotiation is
+	// performed on the first request, after which negotiated is set to "true"
+	// so that subsequent requests do not re-negotiate.
+	autoVersionNegotiation bool
+
+	// vmutex protects the negotiated field
+	vmutex sync.RWMutex
```

Comment:
Wasn't sure how likely it was to need this lock; also thinking of using `sync.Once` instead, but not sure if that would make things clearer

Reply:
Actually; let's simplify, I don't think we need it